### DISCUSSION TITLE
Spin the progress wheel while running a command

### DISCUSSION THREAD

**Discussion by PhillyPhoto** [swiftDialog/swiftDialog#336] [Q&A]
Is there a way to have the "progress" wheel spin continuously like the "wait" wheel does? I'd like to have the wait wheel for things that are pending, and progress for ones that are running to have a visual reference. They are running commands in the background, and incremental percentage changes doesn't really apple.

**Comment by bartreardon** [comment]
nothing built in that does that.

you could create a script that sends increments from 0->100 to the progress indicator in a loop and that would do what you want.

The wait indicator is literally an indeterminate progress and is what it's showing is "something is happening you need to wait for". My suggestion is you don't populate the status indicator at all and instead leave it blank and then use the wait spinner to indicate what item is being processed. you also have status text that can be used as an additional information like "currently installing" or whathaveyou.

**Comment by bartreardon** [comment]
just to illustrate the intent of the indeterminate `wait` icon, this screenshot shows the different uses for the built in status's

<img width="228" alt="image" src="https://github.com/swiftDialog/swiftDialog/assets/3598965/a26ff839-1554-4a04-bbec-ff4aa9be301d">

At the top there are a number of items marked with a status of `success` and descriptive status text. Then follows the current "in progress" item with status text "Installing ..." and the `wait` progress spinner. The remaining items are all yet to be processed and so have "Pending ..." status text and the `pending` status icon applied. Even in a static screenshot, it's clear at a glance which tasks have been completed, which are in progress and which are still to be performed.

The `progress` status icon is intended to be an indicator for an in-progress item where the progress can be measured, otherwise use an indeterminate spinner.

Hope this answers your request. 

As there will be no further work done on this I'll close the issue.
